### PR TITLE
Add constants for wMapConnections bits and group map header wram labels

### PR DESCRIPTION
**Pull request by LinusU**

### CODE REVIEW

File: ram/wram.asm [hunk 1]
```diff
@@ -1804,6 +1804,12 @@ wMapDataPtr:: dw
 wMapTextPtr:: dw
 wMapScriptPtr:: dw
 
+	const_def
```
Comment (Rangi42) [inline ram/wram.asm:1807]: Constants should be defined in constants/\*_constants.asm files. See wram_constants.asm in pokegold and pokecrystal.

**Comment by LinusU** [comment]
@Rangi42 fixed ✅

**Comment by Rangi42** [comment]
Good start! How about using this PR to expand wram_constants.asm to define more of the bit flags? IMO it's easier to deal with than lots of small PRs.

**Comment by LinusU** [comment]
> Good start!

Thanks!

> How about using this PR to expand wram_constants.asm to define more of the bit flags? IMO it's easier to deal with than lots of small PRs.

Unfortunately, this is not something I have time with now. I submitted the changes I had now since I'm working on another project that lead me to investigate how `wMapConnections` worked, and made the change locally.

**Comment by vulcandth** [comment]
Do we know where these wMapConnections bits are being set? Curious if there are other places where we can use these new constants.

**Comment by LinusU** [comment]
> if we are not sure I can spend some time on it.

Please, feel free to! 🙏

**Comment by vulcandth** [comment]
> Do we know where these wMapConnections bits are being set? Curious if there are other places where we can use these new constants.
> 
> Edit: if we are not sure I can spend some time on it.

wMapConnections bits are being loaded directly from the `data/maps/headers/<MAPNAME>.asm` files. Specifically the `map_header` macros.

See the following change suggestions:

https://github.com/pret/pokered/blob/d3128b9a7ffab9d4d9b615acf9527ee2f1744749/ram/wram.asm#L1797-L1811

Everything from wCurMapTileset to wMapConnections should be grouped together and some of the constants should be renamed... Something like:

```asm
wCurMapHeader::
wCurMapTileset:: db
wCurMapWidth:: db
wCurMapDataPtr:: db
wCurMapTextPtr:: db
wCurMapScriptPtr:: db
wCurMapConnections:: db
wCurMapHeaderEnd::

 wNorthConnectionHeader:: map_connection_struct wNorth 
 wSouthConnectionHeader:: map_connection_struct wSouth 
 wWestConnectionHeader::  map_connection_struct wWest 
 wEastConnectionHeader::  map_connection_struct wEast 
```

https://github.com/pret/pokered/blob/d3128b9a7ffab9d4d9b615acf9527ee2f1744749/home/overworld.asm#L2035-L2040

The above should then be changed to:

```asm
 	ld a, [hli] 
 	ld h, [hl] 
 	ld l, a ; hl = base of map header 
 	ld de, wCurMapHeader
 	ld c, wCurMapHeaderEnd - wCurMapHeader
```

**Comment by Rangi42** [comment]
So these are actually the same values as the `EAST/WEST/SOUTH/NORTH` constants, but bits instead of shifted values. pokecrystal declares them together in map_data_constants.asm; I think pokered should do the same.

```
; connection directions
        const_def
        const EAST_F
        const WEST_F
        const SOUTH_F
        const NORTH_F

; wMapConnections
        const_def
        shift_const EAST
        shift_const WEST
        shift_const SOUTH
        shift_const NORTH
```

This would also be the start of porting pokecrystal's `*_F` convention for bitflag constants.

**Review by Rangi42 — APPROVED** [review]

**Pull request merged**